pkg/registry/service/rest.go

```diff
@@ -283,6 +283,21 @@ func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.Rou
 		return nil, nil, errors.NewBadRequest(fmt.Sprintf("invalid service request %q", id))
 	}
 
+	// If a port number was specified...
+	if portNum, err := strconv.ParseInt(portStr, 10, 64); err == nil {
+		// look up the service...
+		if svc, err := rs.registry.GetService(ctx, svcName); err == nil {
```

Comment:
updated to return an error if service lookup fails, or if no service port exists with the specified port number